src/BrowserInterop/Extensions/BrowserInteropJsRuntimeExtension.cs

```diff
@@ -225,11 +226,36 @@ public static async ValueTask<WindowInterop> Window(this IJSRuntime jsRuntime)
 
             var jsRuntimeObjectRef = await jsRuntime.InvokeAsync<JsRuntimeObjectRef>(
                 "browserInterop.callInstanceMethodGetRef",
-                new object[] {windowObject, methodName}.Concat(arguments).ToArray()).ConfigureAwait(false);
+                new object[] { windowObject, methodName }.Concat(arguments).ToArray()).ConfigureAwait(false);
             jsRuntimeObjectRef.JsRuntime = jsRuntime;
             return jsRuntimeObjectRef;
         }
 
+
+        /// <summary>
+        /// Call the method on the js instance and return the references to the items of the array result
+        /// </summary>
+        /// <param name="jsRuntime">Current JS Runtime</param>
+        /// <param name="windowObject">Reference to the JS instance</param>
+        /// <param name="methodName">Method name/path </param>
+        /// <param name="arguments">method arguments</param>
+        /// <returns></returns>
+        public static async ValueTask<JsRuntimeObjectRef[]> InvokeInstanceMethodGetRefs(this IJSRuntime jsRuntime,
+            JsRuntimeObjectRef windowObject, string methodName, params object[] arguments)
+        {
+            if (jsRuntime is null) throw new ArgumentNullException(nameof(jsRuntime));
```

Comment:
nit : I'm not sure this is relevant, better be safe than sorry anyway

Reply:
what do you mean? I had a scenario where a javascript method returns an array of objects that I want to treat as references on the .NET side. Therefore I added that method. Is there another way to do that already?

Reply:
No I meant the null check, the new method is great :)

Reply:
I can remove the null check. Thats ok for me :)